tensorflow/python/ops/math_grad.py

```diff
@@ -309,13 +304,24 @@ def _TanhGrad(op, grad):
     return gen_math_ops._tanh_grad(y, grad)
 
 
+@ops.RegisterGradient("TanhGrad")
+def _TanhGradGrad(op, grad):
+  """Returns grad * (-2 * tanh(x) * (1 - tanh(x) * tanh(x))) """
+  y = op.outputs[0]
+  with ops.control_dependencies([grad.op]):
+    if y.dtype.is_complex:
+      y = math_ops.conj(y)
+    return grad * (-2 * math.ops.tanh(y) *
+                   (1 - math.ops.tanh(y) * math.ops.tanh(y)))
```

Comment:
Looks like this code has never been executed (`math.ops`).  We should definitely have a test using `tf.test.compute_gradient_error`.  Also, CSE out `t = math_ops.tanh(y)`.